src/app/link-list/link-list.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, OnInit, ApplicationRef } from '@angular/core';
+import { LinkService } from '../link.service';
+
+import { Observable, Subject} from 'rxjs';
+import { debounceTime, distinctUntilChanged, switchMap } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-link-list',
+  templateUrl: './link-list.component.html',
+  styleUrls: ['./link-list.component.css']
```

Comment:
Can you add a comment explaining what this component does?
Something like 

"It holds all extracted links and responsible for applying filtering/grouping rules into those links"